docs/tutorials/upgrade-a-chain/sudo-upgrade.md

```diff
@@ -0,0 +1,163 @@
+[storage item](../../knowledgebase/runtime/storage): the ID of the account that has access to the
+pallet's [dispatchable functions](../../knowledgebase/getting-started/glossary#dispatch). The Sudo
+pallet's `sudo` function allows the holder of this account to invoke a dispatchable as the `Root`
+origin. The following pseudo-code demonstrates how this is achieved, refer to the
+[Sudo pallet's source code](https://github.com/paritytech/substrate/blob/v2.0.0-rc6/frame/sudo/src/lib.rs)
+to learn more.
+
+```rust
+fn sudo(origin, call) -> Result {
+  // Ensure caller is the account identified by the administrator key
+  let sender = ensure_signed(origin)?;
+  ensure!(sender == Self::key(), Error::RequireSudo);
+
+  // Dispatch the specified call as the Root origin
+  let res = call.dispatch(Origin::Root);
+  Ok()
+}
+```
+
+## Start the Template Node
+
+Since forkless runtime upgrades do not require network participants to restart their blockchain
+clients, the first step of this tutorial is to start the template node as-is. Build and start the
+unmodified [Node Template](https://github.com/substrate-developer-hub/substrate-node-template).
+
+```shell
+cargo run --release -- --dev --tmp
+```
+
+## Runtime Upgrade Resource Accounting
+
+Dispatchable calls in Substrate are always associated with a
+[weight](../../knowledgebase/learn-substrate/weight), which is used for resource accounting. FRAME's
+System module enforces a
+[`MaximumExtrinsicWeight`](https://substrate.dev/rustdocs/v2.0.0-rc6/frame_system/trait.Trait.html#associatedtype.MaximumExtrinsicWeight)
+and a
+[`MaximumBlockWeight`](https://substrate.dev/rustdocs/v2.0.0-rc6/frame_system/trait.Trait.html#associatedtype.MaximumBlockWeight),
+and the `set_code` function in
+[the System module](https://github.com/paritytech/substrate/blob/v2.0.0-rc6/frame/system/src/lib.rs)
+is intentionally designed to consume the maximum weight that may fit in a block. The `set_code`
+function's weight annotation also specifies that `set_code` is in
+[the `Operational` class](../../knowledgebase/runtime/fees#operational-dispatches) of dispatchable
+functions, which identifies it as relating to network _operations_ and impacts the accounting of its
+resources, such as by exempting it from the
+[`TransactionByteFee`](https://substrate.dev/rustdocs/v2.0.0-rc6/pallet_transaction_payment/trait.Trait.html#associatedtype.TransactionByteFee).
+In order to work within FRAME's safeguards around resource accounting, the Sudo pallet provides the
+[`sudo_unchecked_weight`](https://substrate.dev/rustdocs/v2.0.0-rc6/pallet_sudo/enum.Call.html#variant.sudo_unchecked_weight)
+function, which provides the same capability as the `sudo` function, but accepts an additional
+parameter that is used to specify the (possibly zero) weight to use for the call that is dispatched
+from the `Root` origin. The `sudo_unchecked_weight` function is what will be used to invoke the
```

Comment:
Weight doesn't get passed into downstream calls, only the top-level dispatchable. In this case, the weight of `sudo_unchecked_weight` and whatever is wrapped inside of it get passed as one to the block constructor.
Namely, no weight going in here:
`let res = call.dispatch_bypass_filter(frame_system::RawOrigin::Root.into());`

Reply:
Do you think that anything in this tutorial needs to be updated to clarify this?

Reply:
Well this is not clear:

> accepts an additional ￼parameter that is used to specify the (possibly zero) weight to use for the _call that is dispatched ￼from the `Root` origin._

The call that is dispatched from the root origin doesn't have any weight. I.e., the `_weight` param is for the weight of the `sudo_unchecked_weight` function, including the `call`, but not the weight for the `call`.

Reply:
I'm concerned that you're actually sacrificing clarity for strict technical correctness. Can you suggest a rewording that is as clear and more technically correct?

Reply:
Hmm, I got hung up on this sentence because it felt unclear to which call the `_weight` param applied to. I would just end the sentence earlier:

"...but accepts an additional parameter that is used to specify the (possibly zero) weight to use for the call."